test/unit/test_parser.py

```diff
@@ -1,138 +1,66 @@
+import os
+from glob import glob
+
 import napalm_yang
 
+import pytest
+import yaml
 
-class Tests(object):
 
-    def test_resolve_path_dict(self):
-        data = {
-            "simple": {
-                "path": 123
-            }
-        }
-        path = "simple.path"
-        result = napalm_yang.parsers.base.BaseParser({}, {}).resolve_path(data, path)
-        assert result == 123
+BASE_PATH = os.path.dirname(__file__)
+
+test_parser = [x.split('/')[-1]
+               for x in glob('{}/test_parser/*'.format(BASE_PATH))]
+
 
-    def test_resolve_path_list(self):
-        data = {
-            "simple": [
-                {"path": 123}
-            ]
-        }
-        path = "simple.0.path"
-        result = napalm_yang.parsers.base.BaseParser({}, {}).resolve_path(data, path)
-        assert result == 123
+def to_dict(d):
+    if isinstance(d, dict):
+        d = dict(d)
+        for k, v in d.items():
+            d[k] = to_dict(v)
+    elif isinstance(d, list):
+        r = []
+        for i in d:
+            r.append(to_dict(i))
+        d = r
+    return d
```

Comment:
This is more or less a `copy.deepcopy`, this does not transform the input in a dictionary in any way.